OpenSim/Common/STOFileAdapter.h

```diff
@@ -27,28 +27,141 @@
 
 namespace OpenSim {
 
+class STODataTypeNotSupported : public Exception {
+public:
+    STODataTypeNotSupported(const std::string& file,
+                            size_t line,
+                            const std::string& func,
+                            const std::string& datatype) :
+        Exception(file, line, func) {
+        std::string msg = "Datatype '" + datatype + "' is not supported.";
+
+        addMessage(msg);
+    }
+};
+
+class STODataTypeNotFound : public Exception {
+public:
+    STODataTypeNotFound(const std::string& file,
+                        size_t line,
+                        const std::string& func) :
+        Exception(file, line, func) {
+        std::string msg = "DataType not specified in the header.";
+
+        addMessage(msg);
+    }
+};
+
 /** STOFileAdapter is a DelimFileAdapter that presets the delimiters 
-appropriately for STO files.                                                  */
-class OSIMCOMMON_API STOFileAdapter : public DelimFileAdapter {
+appropriately for STO files. The format of the file is as follows:
+\code
+.................................
+.................................
+........ <metadata> .............
+.................................
+.................................
+endheader
+time colname1 colname2 ..........
+.................................
+.................................
+............ <data-rows> ........
+.................................
+.................................
+\endcode
+For example a TimeSeriesTable_<SimTK::Vec3> could look like:
+\code
+units=m
+DataType=Vec3
+endheader
```

Comment:
>The key-value pairs we insert will be of the format we decide.

I think a convention for key names for metadata that we use would be helpful. We should decide what that should be.  We did the same for the names of properties (e.g. lowercase and underscore separated terms). Since the serialization of metadata is being introduced here I think it is appropriate to discuss/decide here.